### PR TITLE
Implement OctInt/OctString function push down

### PR DESCRIPTION
**Pull request by Ziy1-Tan**
### What problem does this PR solve?

Implement OctInt/OctString function push down

Issue Number: close https://github.com/pingcap/tiflash/issues/5110
Related PR: https://github.com/pingcap/tidb/pull/38316

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Functions/FunctionsString.cpp [hunk 1]
```diff
@@ -5781,6 +5784,219 @@ class FunctionElt : public IFunction
...
+public:
+    static constexpr auto name = "octInt";
+    FunctionOctInt() = default;
+
+    static FunctionPtr create(const Context & /*context*/)
+    {
+        return std::make_shared<FunctionOctInt>();
+    }
+
+    std::string getName() const override { return name; }
```
Comment (ywqzzy) [inline dbms/src/Functions/FunctionsString.cpp:5798]:
```suggestion
    String getName() const override { return name; }
```

File: dbms/src/Functions/FunctionsString.cpp [hunk 2]
```diff
@@ -5781,6 +5784,219 @@ class FunctionElt : public IFunction
...
+        if (col == nullptr)
+        {
+            return false;
+        }
+        size_t size = col->size();
+
+        auto col_res = ColumnString::create();
+
+        ColumnString::Chars_t & res_chars = col_res->getChars();
+        // Convert a UInt64 to hex, will cost 17 bytes at most
```
Comment (ywqzzy) [inline dbms/src/Functions/FunctionsString.cpp:5855]: hex?

File: dbms/src/Functions/FunctionsString.cpp [hunk 3]
```diff
@@ -5781,6 +5784,219 @@ class FunctionElt : public IFunction
...
+public:
+    static constexpr auto name = "octString";
+    FunctionOctString() = default;
+
+    static FunctionPtr create(const Context & /*context*/)
+    {
+        return std::make_shared<FunctionOctString>();
+    }
+
+    std::string getName() const override { return name; }
```
Comment (ywqzzy) [inline dbms/src/Functions/FunctionsString.cpp:5962]:
```suggestion
    String getName() const override { return name; }
```

File: dbms/src/Functions/FunctionsString.cpp (hunk 2 above)
Comment (Ziy1-Tan) [reply dbms/src/Functions/FunctionsString.cpp]: oops, It should be oct.

Comment (ywqzzy) [reply dbms/src/Functions/FunctionsString.cpp]: > oops, It should be oct.

And we need 2 LGTM to merge the pr.

File: dbms/src/Functions/tests/gtest_strings_octint.cpp [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+TEST_F(OctIntTest, Simple)
+try
+{
+    ASSERT_COLUMN_EQ(
+        toNullableVec<String>({"0",
+                               "1",
+                               "2",
+                               "4",
+                               "10"}),
+        executeFunction(
```
Comment (ywqzzy) [inline dbms/src/Functions/tests/gtest_strings_octint.cpp:39]: Maybe add some test for const column?

Comment (Ziy1-Tan) [reply dbms/src/Functions/tests/gtest_strings_octint.cpp:39]: let me take a look.

**Pull request closed without merging**